### PR TITLE
feat: Added Course review Api

### PR DESCRIPTION
**Pull request by zamanafzal**
JIRA: https://2u-internal.atlassian.net/browse/ENT-6872

Added an API to get the course review model data.

### CODE REVIEW

File: course_discovery/apps/api/serializers.py [hunk 1]
```diff
@@ -2624,3 +2624,19 @@ def get_field_info(self, field):
             return self.create_type_options(info)
 
         return info
+
+
+class CourseReviewSerializer(serializers.ModelSerializer):
+    most_common_goal = serializers.CharField(source='get_most_common_goal_display')
```
Comment (sameenfatima78) [inline course_discovery/apps/api/serializers.py:2630]: Can't track the source, where are we getting this value from?

File: course_discovery/apps/api/v1/views/courses.py [hunk 2]
```diff
@@ -522,3 +522,15 @@ class CourseRecommendationViewSet(RetrieveModelMixin, viewsets.GenericViewSet):
...
+class CourseReviewViewSet(viewsets.ReadOnlyModelViewSet):
+    """ CourseReview. """
+    queryset = CourseReview.objects.all()
+    serializer_class = serializers.CourseReviewSerializer
+    permission_classes = (IsAuthenticated,)
+    lookup_field = 'course_key'
+
+    def retrieve(self, request, *args, **kwargs):
+        """ Retrieve details for a CourseReview. """
+        return super().retrieve(request, *args, **kwargs)
```
Comment (sameenfatima78) [inline course_discovery/apps/api/v1/views/courses.py]: nit: this code can be removed

**Review by sameenfatima78 — APPROVED** [review]
a few nits, otherwise looks good!

File: course_discovery/apps/api/serializers.py (hunk 1 above)
Comment (zamanafzal) [reply course_discovery/apps/api/serializers.py:2630]: By using `get_most_common_goal_display` I'm returning the `most_common_goal` value except for the key. This is the default functionality of Django for choice fields.

File: course_discovery/apps/api/v1/views/courses.py [hunk 3]
```diff
@@ -522,3 +522,11 @@ class CourseRecommendationViewSet(RetrieveModelMixin, viewsets.GenericViewSet):
     permission_classes = (IsAuthenticated, IsCourseEditorOrReadOnly,)
     serializer_class = serializers.CourseWithRecommendationsSerializer
     queryset = serializers.MinimalCourseSerializer.prefetch_queryset()
+
+
+class CourseReviewViewSet(viewsets.ReadOnlyModelViewSet):
+    """ CourseReview. """
+    queryset = CourseReview.objects.all()
+    serializer_class = serializers.CourseReviewSerializer
+    permission_classes = (IsAuthenticated,)
```
Comment (DawoudSheraz) [inline course_discovery/apps/api/v1/views/courses.py]: Please verify what other permissions are needed here. This is only requiring auth and nothing else.

File: course_discovery/apps/api/v1/views/courses.py [hunk 4]
```diff
@@ -522,3 +522,11 @@ class CourseRecommendationViewSet(RetrieveModelMixin, viewsets.GenericViewSet):
     permission_classes = (IsAuthenticated, IsCourseEditorOrReadOnly,)
     serializer_class = serializers.CourseWithRecommendationsSerializer
     queryset = serializers.MinimalCourseSerializer.prefetch_queryset()
+
+
+class CourseReviewViewSet(viewsets.ReadOnlyModelViewSet):
```
Comment (DawoudSheraz) [inline course_discovery/apps/api/v1/views/courses.py]: nit: Please move this view to its own file.

File: course_discovery/apps/api/v1/tests/test_views/test_courses.py [hunk 5]
```diff
@@ -2239,3 +2241,34 @@ def test_recommendations(self):
...
+            reviews_count=100,
+            avg_course_rating=4.5,
+            confident_learners_percentage=0.85,
+            most_common_goal=CourseReview.JOB_ADVANCEMENT,
+            most_common_goal_learners_percentage=0.9,
+            total_enrollments=500
+        )
+
+    def test_retrieve_course_review(self):
+        url = f'/api/v1/course-review/{self.course_review.course_key}/'
```
Comment (DawoudSheraz) [inline course_discovery/apps/api/v1/tests/test_views/test_courses.py]: nit:  use reverse() to create the url.

File: course_discovery/apps/api/v1/views/courses.py (hunk 3 above)
Comment (zamanafzal) [reply course_discovery/apps/api/v1/views/courses.py]: We don't need any other permissions here. This data will be shown to the user along with other course details like **course title**, and **course description**.

**Review by irfanuddinahmad — APPROVED** [review]

**Review by AliAdnanSohail — APPROVED** [review]

**Review by AfaqShuaib09 — APPROVED** [review]

**Pull request merged**